src/module.c

```diff
@@ -455,17 +464,22 @@ DLLEXPORT void jl_set_current_module(jl_value_t *m)
     jl_current_module = (jl_module_t*)m;
 }
 
-DLLEXPORT jl_value_t *jl_module_usings(jl_module_t *m)
-{
-    jl_array_t *a = jl_alloc_array_1d(jl_array_any_type, 0);
-    JL_GC_PUSH1(&a);
-    for(int i=(int)m->usings.len-1; i >= 0; --i) {
-        jl_array_grow_end(a, 1);
-        jl_module_t *imp = (jl_module_t*)m->usings.items[i];
-        jl_cellset(a,jl_array_dim0(a)-1, (jl_value_t*)imp);
+DLLEXPORT jl_value_t *jl_module_usings(jl_module_t *mod)
+{
+    jl_array_t *arr = jl_alloc_array_1d(jl_array_any_type, 0);
+    size_t len = mod->usings.len;
+    if (len) {
+        JL_GC_PUSH1(&arr);
+        // Why doesn't this grow the array all at once?
+        // jl_array_grow_end(arr, len);
+        jl_module_t **usingp = (jl_module_t **)mod->usings.items;
+        do {
+            jl_array_grow_end(arr, 1);
+            jl_cellset(arr,jl_array_dim0(arr)-1, *usingp++);
+        } while (--len);
+        JL_GC_POP();
     }
-    JL_GC_POP();
-    return (jl_value_t*)a;
+    return (jl_value_t*)arr;
 }
```

Comment:
@ScottPJones, before you continue to modify / work on this, please consider splitting it up into separate logical commits that each address as specific an issue as possible.